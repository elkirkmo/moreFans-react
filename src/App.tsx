import React, { useState, useEffect } from 'react';
import Team from './components/Team';
import data from './data/nfl';
import './App.css';

interface Props {
  className: string;
  fanCount: string;
  id: string;
  mascot: string;
  // mascotFacts: {
  //   number: string;
  // };
  mythical: boolean;
  name: string;
  vs: boolean;
}

function App() {
  const [currentTeam, setCurrentTeam] = useState(0);
  return (
    <div className="App">
      <header>
        <h1>More Fans</h1>
        <Team {...data[currentTeam]} />
        {currentTeam > 0 && (
          <button onClick={() => setCurrentTeam(currentTeam - 1)}>
            Previous Team
          </button>
        )}
        <button onClick={() => setCurrentTeam(currentTeam + 1)}>
          Next Team
        </button>
      </header>
    </div>
  );
}

export default App;
