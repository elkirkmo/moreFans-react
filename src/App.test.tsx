import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders the application', () => {
  const { getByText, asFragment } = render(<App />);
  const linkElement = getByText(/Chicago/i);
  expect(linkElement).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
