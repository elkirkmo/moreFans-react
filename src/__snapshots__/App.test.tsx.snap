// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the application 1`] = `
<DocumentFragment>
  <div
    class="App"
    id="nfl"
  >
    <header>
      <h1>
        More Fans
      </h1>
    </header>
    <div
      class="chi"
      id="1"
    >
      <h2>
        Chicago
      </h2>
      <p>
        Real-life Fans
      </p>
      <p>
        12,092,476
      </p>
      <h3>
        Mascot
      </h3>
      <p>
        Bears
      </p>
      <p>
        1,148,364
      </p>
    </div>
    <button>
      Next Team
    </button>
  </div>
</DocumentFragment>
`;
